lib/src/iterable_extensions.dart

```diff
@@ -0,0 +1,519 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// Extension methods on iterables.
+
+import 'package:collection/src/utils.dart';
+
+import "algorithms.dart";
+
+/// Extensions that apply to all iterables.
+extension IterableExtension<T> on Iterable<T> {
+
+  /// The elements that do not satisfy [test].
+  Iterable<T> whereNot(bool test(T element)) =>
+      where((element) => !test(element));
+
+  /// Creates a sorted list of the elements of the iterable.
+  ///
+  /// The elements are ordered by the [compare] [Comparator].
+  List<T> sorted(int compare(T a, T b)) => [...this]..sort(compare);
```

Comment:
Should we match the signature on `List.sort` and make the argument optional?

Reply:
The signature of `List.sort` is actually bad. It allows you to omit the `compare`, but that only really works when the elements are `Comparable`. With extension methods we can do *better*.
There is a `sorted([int compare(E a, E b)])` on `Iterable<T extends Comparable<T>>`.